commandbased/flowcontrol.py

```diff
@@ -3,16 +3,102 @@
 intuitive. For more information, check each method's docstring.
 """
 
-from wpilib.command import Command, CommandGroup
-from commandbased.conditionalcommand import ConditionalCommand
+from wpilib.command import Command, CommandGroup, ConditionalCommand
 from commandbased.cancelcommand import CancelCommand
 
 import inspect
 
-__all__ = ["IF", "ELIF", "ELSE", "WHILE", "RETURN", "BREAK"]
+__all__ = ["IF", "ELIF", "ELSE", "WHILE", "RETURN", "BREAK", "CommandFlow"]
 
 
-def _getCommandGroup():
+class ConditionalFlow(ConditionalCommand):
+    def _condition(self):
+        if self.flowCondition:
+            return self.flowCondition()
+
+        return False
+
+
+class CommandFlow(CommandGroup):
+    def __init__(self, name):
+        CommandGroup.__init__(self, name)
+
+        callingFlow = _getCommandFlow()
+        self._source = getattr(callingFlow, '_source', self)
+        self._ifStack = None
+
+
+    def _popIfStack(self):
+        """
+        We buffer conditionals until the last moment so we don't have trouble with
+        Commands being locked when they're added to a CommandGroup.
+        """
+        if self._ifStack:
+            print(self._ifStack)
+            top = self._ifStack.pop(0)
+            print(f'Building stack for {top[1].getName()}')
+            cmd = None
+            for x in reversed(self._ifStack):
+                if x[0]:
+                    cmd = ConditionalFlow("flowcontrolELIF", x[1], cmd)
+                    cmd.flowCondition = x[0]
```

Comment:
This seems a bit odd. Is there a particular reason to not pass in the `flowCondition` to the constructor?